app/extras/model_locator.rb

```diff
@@ -5,6 +5,8 @@ def locate
       resource_class.find id_param
     elsif key_param = params[:"#{model}_key"]
       resource_class.find_by_key key_param
+    elsif model == :user
```

Comment:
hrrrmm that's really good. Makes me think we should not have a key on user but rather use username ..